modules/plugin/geopkg/src/main/java/org/geotools/geopkg/GeoPkgDialect.java

```diff
@@ -642,7 +644,7 @@ protected void addSupportedHints(Set<Hints.Key> hints) {
             List<Class> resultTypes = maybeResultTypes.get();
             if (resultTypes.size() == 1) {
                 Class targetType = resultTypes.get(0);
-                if (Date.class.isAssignableFrom(targetType)) {
+                if (java.util.Date.class.isAssignableFrom(targetType)) {
```

Comment:
This check looked kind of odd when I first saw it; I guess this dialect knows it is only doing stilly things for Date.